feat: added kratos for authentication with username + password

siddhart1o1 opened the pull request:
+ added kratos configuration.
+ implemented login/registration on client side.
+ added context API to manage session.
+ minor changes.
    + fixed 12 digit pasting issue on voucher page.
    + merchant cannot access the voucher from their voucher history (some more changes required from API side).
+ changes in navbar for logged in and logged-out users.

siddhart1o1 committed:
feat: authentication with kratos : username and password

siddhart1o1 committed:
Merge remote-tracking branch 'origin' into 28-add-kratos-for-authentication-with-username-+-password

siddhart1o1 committed:
fix: changes in navbar for logged-in users

siddhart1o1 committed:
fix:changes in voucher input and user link not clickable

siddhart1o1 committed:
remove oathkeeper

siddhart1o1 committed:
Merge branch 'siddhart1o1-28-add-kratos-for-authentication-with-username-+-password'

nicolasburtey commented on config/kratos/identity.schema.json:
```diff
@@ -0,0 +1,29 @@
+          "type": "string",
+          "format": "string",
+          "title": "username",
+          "minLength": 8,
```
this could be 3 here

nicolasburtey commented on config/kratos/identity.schema.json:
```diff
@@ -0,0 +1,29 @@
+          "format": "string",
+          "title": "username",
+          "minLength": 8,
+          "maxLength": 30,
```
and maybe 18 max?

nicolasburtey commented on config/kratos/kratos.yml:
```diff
@@ -0,0 +1,84 @@
+    - id: default
+      url: file:///etc/config/kratos/identity.schema.json
+
+courier:
+  smtp:
+    connection_uri: smtps://test:test@mailslurper:1025/?skip_ssl_verify=true
```
probably not needed with username only

siddhart1o1 replied:
I tried removing it earlier, but I think this is a required property.
```
The configuration contains values or keys which are invalid:
courier.smtp.connection_uri: <nil>
                             ^-- one or more required properties are missing
time=2023-08-11T13:42:24Z level=error msg=Unable to instantiate configuration. audience=application error=map[message:I[#/courier/smtp] S[#/properties/courier/properties/smtp/required] missing properties: "connection_uri" stack_trace:stack trace could not be recovered from error type *jsonschema.ValidationError] service_name=Ory Kratos service_version=v0.13.0
Error: I[#/courier/smtp] S[#/properties/courier/properties/smtp/required] missing properties: "connection_uri"
Usage:
  kratos serve [flags]
Flags:
  -c, --config strings   Path to one or more .json, .yaml, .yml, .toml config files. Values are loaded in the order provided, meaning that the last config file overwrites values from the previous config file.
      --dev              Disables critical security features to make development easier
  -h, --help             help for serve
      --sqa-opt-out      Disable anonymized telemetry reports - for more information please visit https://www.ory.sh/docs/ecosystem/sqa
      --watch-courier    Run the message courier as a background task, to simplify single-instance setup
I[#/courier/smtp] S[#/properties/courier/properties/smtp/required] missing properties: "connection_uri"
```

nicolasburtey commented on package-lock.json:
```diff
@@ -23,6 +23,9 @@
         "@headlessui/react": "^1.7.14",
         "@mui/icons-material": "^5.11.16",
         "@mui/material": "^5.13.5",
+        "@ory/client": "^1.1.41",
```
you probably have some command line that keep using `npm`? let's stick to yarn for now.

siddhart1o1 replied:
These are older commits now we are using yarn.

nicolasburtey commented on app/voucher/page.tsx:
```diff
@@ -30,12 +30,58 @@ export default function VoucherPage() {
-      ...prevState,
-      [e.target.name]: e.target.value,
-    }));
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
```
ideally we want to have one PR / one improvement.

siddhart1o1 committed:
git checkout -b 'siddhart1o1-28-add-kratos-for-authentication-with-username-+-password' main
git pull https://github.com/siddhart1o1/galoy-voucher.git '28-add-kratos-for-authentication-with-username-+-password'

siddhart1o1 committed:
conflic changes

siddhart1o1 committed:
change: username length
